iree/compiler/Conversion/LinalgToNVVM/KernelConfig.cpp

```diff
@@ -24,57 +24,22 @@ using namespace mlir::iree_compiler;
 
 static constexpr unsigned cudaWarpSize = 32;
 
-/// Fills `inputTypes` and `outputTypes` with the original input/output types
```

Comment:
I think its fine to keep this. Its a pattern that should be easy to replace to use `linalg.tile` when we can. I am just cleaning up the `concretize...` pass and wondering why we have this attribute mechanism. Actually we should be able to use the `ViewLikeInterface` to get the source already. I am cleaning that up on the SPIR-V side. You could then adapt this?

Reply:
I removed it because I realized it was currently not working. On the SPIRV path it relies on concretize pass to go up the ssa chain, analyse the subview ops and then set those attributes (somehow I thought this was set in flow). It felt a bit too much to pull all this logic so I would rather just wait for linalg.tile to be ready. What do you think? If the logic is simpler after your refactoring maybe I can match it, but I would rather avoid adding a fragile analyse of the subview op.

Reply:
I sent you some WIP that I am trying to solve the similar problem on SPIR-V side. We actually dont need the attribute and traversal. All the ops we need to traverse implement a ViewOpInterface (something like this https://github.com/llvm/llvm-project/blob/d61b40ed27509d8a99b4d85499a8d5ca9f37f111/mlir/lib/Dialect/Linalg/Analysis/DependenceAnalysis.cpp#L69) to not use the attribute based hand-shake which is very shaky. I was going to try this out today. If you want you can wait for me to finish that up and use that instead

Reply:
Interesting, it still requires going up the dependency chain to find the first subview op? I don't understand how else we can handle this since the original shape information is not in the tiled op anymore.
I won't really need this for a long time I think so that's why I would rather avoid going for an intermediate solution.

Reply:
`SubView` implements the `ViewOpInterface`. It returns the source. and you can recurse to get the first subview op.

Reply:
So you are saying, we can just have a function that recurse through the subViewOps until it finds the original memref? Yeah that makes sense. Thanks for sharing the code, I think I can apply just copy what you have. I would rather keep this change for a future PR in order to not add extra complexity to this one if this is okay with you.

Reply:
Sounds good. Let me do a quick review.